.github/workflows/validate-theme-entry.yml

```diff
@@ -0,0 +1,269 @@
+                }
+
+                if (/^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/.test(theme.author)) {
+                    addWarning('We generally discourage from including email addresses in the `author` field.');
+                }
+
+                if (theme.name.toLowerCase().includes('obsidian')) {
+                    addWarning(`We discourage themes from including the word "Obsidian" in their name since it's redundant and makes the theme selection screen harder to visually parse.`);
+                }
+
+                if (themes.filter(t => t.name === theme.name).length > 1) {
+                    addError('There is already a theme with this name');
+                }
+
+                if (themes.filter(t => t.repo === theme.repo).length > 1) {
+                    addError('There is already a entry pointing to the `' + theme.repo + '` repository');
+                }
+
+
+                try {
+                    await github.rest.repos.getContent({
+                        owner, repo, path: 'theme.css'
+                    });
+                } catch (e) {
+                    addError('Your repository does not include a `theme.css` file');
+                }
+
+                try {
+                    await github.rest.repos.getContent({
+                        owner, repo, path: 'obsidian.css'
+                    });
+                    addWarning('Your repository includes a `obsidian.css` file, this is only used in legacy versions of Obsidian.');
+                } catch (e) { }
+
+                try {
+                    const screenshot = await github.rest.repos.getContent({
+                        owner, repo, path: theme.screenshot
+                    });
+                    const imageMeta = await probe(screenshot.data.download_url);
+                    if (imageMeta.type !== 'png' && imageMeta.type !== 'jpg') {
+                        addError('Theme screenshot is not of filetype `.png` or `.jpg`');
+                    }
+                    const recommendedSize = 'we generally recommend a size around 512 × 288 pixels';                    
+                    if (imageMeta.width > 1000 || imageMeta.height > 500) {
+                        addError(`Your theme screenshot is too big, ${recommendedSize}`);
+                    }
+                    else if (imageMeta.width < 250 || imageMeta.height < 100) {
+                        addError(`Your theme screenshot is too small, ${recommendedSize}`);
+                    } else if (imageMeta.width !==512 || image.height !== 288) {
+                        addWarning(`Theme theme screenshot size is not optimal, ${recommendedSize}`);
+                    }
+                } catch (e) {
+                    addError('The theme screenshot cannot be found.');
+                }
+
+                // only validate releases if version is included
+                if (manifest.hasOwnProperty('version')) {
```

Comment:
Isn't `version` a required key?

Reply:
According to the docs yes, but not according to the code.